compiler/rustc_hir/src/hir.rs

```diff
@@ -3181,7 +3181,7 @@ impl<'hir> Item<'hir> {
 }
 
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
-#[derive(Encodable, Decodable, HashStable_Generic)]
+#[derive(Encodable, Decodable, HashStable_Generic, TypeFoldable, TypeVisitable)]
```

Comment:
why is that needed?

Reply:
Similar to `BoundConstness`, this exists because `Unsafety` also impls `Relate` (used in `<FnSig as Relate>::relate`).

Reply:
Added explanation in 8834fbf4346d25f3051044979379897299ca5e7f